### PR TITLE
Issue 562: fix clang warnings

### LINKED ISSUE #562: clang warnings in macOS tests
There is a lot of warnings in macOS clang built. It would be to fix important ones and ignore explicitly others.

### PR DESCRIPTION
**Pull request by jkotan**
It resolves #562 by fixing or ignoring clang warnings in macOS built.

### CODE REVIEW

File: src/h5cpp/node/dataset.hpp [hunk 1]
```diff
@@ -566,7 +567,7 @@ class DLL_EXPORT Dataset : public Node
     filter::ExternalFilters filters() const;
 
   private:
-    datatype::Datatype file_type;
+    datatype::Datatype file_type_;
```
Comment (yuelongyu) [inline src/h5cpp/node/dataset.hpp:570]: why does the variable name has to be changed?

**Review by yuelongyu — COMMENTED** [review]
There are multiple places using 
```
unsigned2signed or signed2unsigned
```
what's the difference between these methods and static_cast, can they be interchangeable?

File: src/h5cpp/node/dataset.hpp (hunk 1 above)
Comment (jkotan) [reply src/h5cpp/node/dataset.hpp:570]: I change the class variable because it is the same as a write method parameter and clang report it as a warning.

**Comment by eugenwintersberger** [comment]
Looks good to me. Go for the merge.

**Review by eugenwintersberger — APPROVED** [review]
Looks good to me. Go for the merge.

**Pull request merged**